common/types/block_test.go

```diff
@@ -42,27 +42,29 @@ func TestFields(t *testing.T) {
 
 func TestStringToNodeID(t *testing.T) {
 	pubkey := genByte32()
+	pubkey2 := genByte32()
 	nodeID1 := NodeID{
 		Key:          util.Bytes2Hex(pubkey[:]),
-		VRFPublicKey: []byte("22222"),
+		VRFPublicKey: []byte(nil),
 	}
 	nodeIDStr := nodeID1.String()
 	reversed, err := StringToNodeID(nodeIDStr)
 
 	r := require.New(t)
 	r.NoError(err, "Error converting string to NodeID")
 	r.Equal(nodeID1.Key, reversed.Key, "Node ID deserialization Key does not match")
-	r.Equal(nodeID1.VRFPublicKey, reversed.VRFPublicKey, "Node ID deserialization VRF Key does not match")
 
 	// Test too short
 	_, err = StringToNodeID(string(pubkey[:10]))
 	r.Error(err, "Expected error converting too-short string to NodeID")
 
-	// Test too long
-	var x [129]byte
-	rand.Read(x[:])
-	_, err = StringToNodeID(string(x[:]))
-	r.Error(err, "Expected error converting too-long string to NodeID")
+	//Test too long : make sure that everything after the first 64 characters is ignored
+	longString := nodeID1.String() + string(pubkey2[:])
+	r.Greater(len(longString), len(nodeID1.String()))
+	reversed, err = StringToNodeID(longString)
+	r.NoError(err, "Error converting too-long string to NodeID")
+	r.Equal(nodeID1.Key, reversed.Key, "NodeID does not match for too-long string")
+	r.Equal([]byte{}, reversed.VRFPublicKey, "VRF Key is not empty for too-long string")
```

Comment:
"is" empty, not "is not" empty, right?

Reply:
also, is there a difference between `[]byte{}` and `[]byte(nil)` (which you use below)?